src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -535,57 +618,68 @@ private void updateParticipantMetadata(
         uniqueDates.addAll( entity.getOrDefault( RECORDED_DATE_TIME_FQN, Set.of() ) );
         metadataEntityData.put( propertyTypeIdsByFQN.get( RECORDED_DATE_TIME_FQN ), uniqueDates );
 
-        dataApi.updateEntitiesInEntitySet( entitySetIdMap.get( METADATA_ENTITY_SET_NAME ),
+        dataApi.updateEntitiesInEntitySet( metadataESID,
                 ImmutableMap.of( metadataEntityKeyId, metadataEntityData ),
                 UpdateType.PartialReplace );
 
         Map<UUID, Set<Object>> hasEntityData = new HashMap<>();
         hasEntityData.put( propertyTypeIdsByFQN.get( OL_ID_FQN ), Set.of( firstDateTime ) );
-        UUID hasEntityKeyId = reserveHasEntityKeyId( metadataEntityData, dataIntegrationApi );
-        dataApi.updateEntitiesInEntitySet( entitySetIdMap.get( HAS_ENTITY_SET_NAME ),
+        UUID hasEntityKeyId = reserveHasEntityKeyId( hasESID, metadataEntityData, dataIntegrationApi );
+        dataApi.updateEntitiesInEntitySet( hasESID,
                 ImmutableMap.of( hasEntityKeyId, hasEntityData ),
                 UpdateType.PartialReplace );
 
-        EntityDataKey dst = new EntityDataKey( entitySetIdMap.get( METADATA_ENTITY_SET_NAME ), metadataEntityKeyId );
-        EntityDataKey edge = new EntityDataKey( entitySetIdMap.get( HAS_ENTITY_SET_NAME ), hasEntityKeyId );
+        EntityDataKey dst = new EntityDataKey( metadataESID, metadataEntityKeyId );
+        EntityDataKey edge = new EntityDataKey( metadataESID, hasEntityKeyId );
         EntityDataKey src = new EntityDataKey( participantEntitySetId, participantEntityKeyId );
         DataEdgeKey dataEdgeKey = new DataEdgeKey( src, dst, edge );
         dataApi.createEdges( Set.of( dataEdgeKey ) );
 
         logger.info( "Uploaded user metadata entries: participantId = {}", participantId );
     }
 
+    private UUID ensureEntitySetExists( UUID organizationId, String appName, String template ) {
```

Comment:
@katherinebernstein I'm fairly confident that this isn't how you should pull entity set names from apps/v2

Reply:
This seems correct, under the assumption that chronicle server loads app configs based on the chronicle@openlattice.com user having access to various app configs. Generally speaking, an entity type collection defines various named entity types (i.e. `study`, `device`, `participant`) as template types, and an entity set collection will map those template types onto entity set ids. It looks like this is loading those mappings up front and using this logic (through getEntitySetId) to load an entity set id for a particular organization's config.

A few potential issues / ways to make this clearer:

- I think the various template types should be stored as an enum, rather than free-floating strings. That will help constrain them to a defined set of values and prevent any accidental passing of invalid type names here.
- The `entitySetIdsByOrgId` map appears to be a map from `appName -> organizationId -> templateTypeName -> entitySetId`. We should only ever be referencing one app here (chronicle) so we should be able to remove that outer `appName` mapping which should simplify a lot of logic here
- The `entitySetIdsByOrgId` map is only loaded at startup, which may present problems. If an entity set collection mapping is updated (ie an entity set is swapped out for a different one), or if chronicle is installed for a new organization, those changes won't be picked up. Reloading the mappings on every call feels potentially excessive, but maybe we can cache + reload them at some interval or something like that.

Reply:
Good points @katherinebernstein, and thanks for clearing things up.

On your second bullet point though: 
> The entitySetIdsByOrgId map appears to be a map from appName -> organizationId -> templateTypeName -> entitySetId. We should only ever be referencing one app here (chronicle) so we should be able to remove that outer appName mapping which should simplify a lot of logic here

When I synced with @UnsungHero97 we thought it would be best to split chronicle into 3 components: the core features (study, participants, etc), data collection (for the android app usage), and surveys (ol.question, ol.answer, ol.survey etc). The idea was to ensure that organizations only have the entity sets that they need. For example, organizations that are not using the survey feature don't need to have ol.answer, ol.survey, ol.respondswith entity sets. The overhead however is that each app config creates a new set of roles to manage. 

Thoughts?

Reply:
So template type name resolves to an entity set id, which is stored in the app config?

Reply:
basically -- `template type -> entity set id` is stored in the entity set collection. The app config points to an entity set collection.

Reply:
@anzioka that makes sense -- although in that case I would suggest that the 3 app components be moved into an enum as well, instead of passing around strings